test/NITTA/Intermediate/Simulation/Tests.hs

```diff
@@ -36,7 +38,22 @@ simulationTests =
                 l1 = loop 0 "b2" ["a1"]
                 l2 = loop 1 "c" ["b1", "b2"]
                 a = add "a1" "b1" ["c"]
-            mapM_ (([l1, l2, a] @=?) . f) $ permutations [l1, l2, a]
+            mapM_ (([l1, l2, a] @=?) . f) [[l1, l2, a], [l1, a, l2], [a, l1, l2]]
```

Comment:
Why?

Reply:
Please, rename `f`. It is very bad naming…

Reply:
l1 and l2 can be executed on the same wave, so we keep the execution order from the source list.
When you use permutations you also get lists where l2 is placed before l1 ([l2,l1,a], [l2,a,l1]) so the output will be [l2,l1,a] which is not equal to [l1 ,l2,a]